x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/services/elastic/authorization/AuthorizationPoller.java

```diff
@@ -0,0 +1,261 @@
+    }
+
+    private AuthorizationPoller(TaskFields taskFields, Parameters parameters) {
+        this(
+            taskFields,
+            parameters.serviceComponents,
+            parameters.authorizationRequestHandler,
+            parameters.sender,
+            parameters.elasticInferenceServiceSettings,
+            parameters.modelRegistry,
+            parameters.client,
+            null
+        );
+    }
+
+    // default for testing
+    AuthorizationPoller(
+        TaskFields taskFields,
+        ServiceComponents serviceComponents,
+        ElasticInferenceServiceAuthorizationRequestHandler authorizationRequestHandler,
+        Sender sender,
+        ElasticInferenceServiceSettings elasticInferenceServiceSettings,
+        ModelRegistry modelRegistry,
+        Client client,
+        // this is a hack to facilitate testing
+        Runnable callback
+    ) {
+        super(taskFields.id, taskFields.type, taskFields.action, taskFields.description, taskFields.parentTask, taskFields.headers);
+        this.serviceComponents = Objects.requireNonNull(serviceComponents);
+        this.authorizationHandler = Objects.requireNonNull(authorizationRequestHandler);
+        this.sender = Objects.requireNonNull(sender);
+        this.elasticInferenceServiceSettings = Objects.requireNonNull(elasticInferenceServiceSettings);
+        this.elasticInferenceServiceComponents = new ElasticInferenceServiceComponents(
+            elasticInferenceServiceSettings.getElasticInferenceServiceUrl()
+        );
+        this.modelRegistry = Objects.requireNonNull(modelRegistry);
+        this.client = new OriginSettingClient(Objects.requireNonNull(client), ClientHelper.INFERENCE_ORIGIN);
+        this.callback = callback;
+    }
+
+    public void start() {
+        if (initialized.compareAndSet(false, true)) {
+            logger.debug("Initializing EIS authorization logic");
+            serviceComponents.threadPool().executor(UTILITY_THREAD_POOL_NAME).execute(this::scheduleAndSendAuthorizationRequest);
+        }
+    }
+
+    /**
+     * This should only be used for testing to wait for the first authorization response to be received.
+     */
+    public void waitForAuthorizationToComplete(TimeValue waitTime) {
+        try {
+            if (receivedFirstAuthResponseLatch.await(waitTime.getSeconds(), TimeUnit.SECONDS) == false) {
+                throw new IllegalStateException("The wait time has expired for first authorization response to be received.");
+            }
+        } catch (InterruptedException e) {
+            throw new IllegalStateException("Waiting for first authorization response to complete was interrupted");
+        }
+    }
+
+    @Override
+    protected void onCancelled() {
+        shutdown();
+        markAsCompleted();
+    }
+
+    // default for testing
+    void shutdown() {
+        shutdown.set(true);
+        if (lastAuthTask.get() != null) {
+            lastAuthTask.get().cancel();
+        }
```

Comment:
i don't think it matters since `scheduleAndSendAuthorizationRequest` checks the shutdown status, but in theory one thread could set a different  `ScheduledCancellable` in between 145 and 146
```suggestion
        var authTask = lastAuthTask.get();
        if (authTask != null) {
            authTask.cancel();
        }
```